internal/handlers/auth/user_auth_handler.go

```diff
@@ -0,0 +1,54 @@
+package auth
+
+import (
+	"fmt"
+	"github.com/antonioo83/license-server/config"
+	"github.com/antonioo83/license-server/internal/models"
+	"github.com/antonioo83/license-server/internal/repositories/interfaces"
+	"net/http"
+	"strings"
+)
+
+const Admin = 1
+const User = 2
+
+type UserAuth struct {
+	Role int
+	User models.User
+}
+
+type UserAuthHandler struct {
+	userRepository interfaces.UserRepository
+	config         config.Config
+}
```

Comment:
Так вот, по поводу сегрегации интерфесов: в этом пакете мы используем только один метод из всех определенных в interfaces.UserRepository, поэтому мы можем  примерно так
```suggestion

type TokenProvider interface {
        FindByToken(code string) (*models.User, error)
}

type UserAuthHandler struct {
	userRepository TokenProvider
	config         config.Config
}
// Думаю дальше с конструктором понятно.
```
Преимущество в том, что больше никто не использует этот специфичный метод, а мы не используем остальные методы репозитори, следовательно прокидывая какую-либо имплементацию провайдера токенов, нам важен только один метод которые мы используем.  С моей точки зрения так проще все смотрится.